homeassistant/components/drop_connect/sensor.py

```diff
@@ -0,0 +1,432 @@
+                DROP_CurrentSystemPressureSensor(coordinator),
+            ]
+        )
+    elif device_type == DEV_LEAK_DETECTOR:
+        entities.extend(
+            [
+                DROP_BatterySensor(coordinator),
+                DROP_TemperatureSensorC(coordinator),
+                DROP_TemperatureSensorF(coordinator),
+            ]
+        )
+    elif device_type == DEV_PROTECTION_VALVE:
+        entities.extend(
+            [
+                DROP_BatterySensor(coordinator),
+                DROP_CurrentFlowRateSensor(coordinator),
+                DROP_CurrentSystemPressureSensor(coordinator),
+                DROP_TemperatureSensorC(coordinator),
+                DROP_TemperatureSensorF(coordinator),
+            ]
+        )
+    elif device_type == DEV_PUMP_CONTROLLER:
+        entities.extend(
+            [
+                DROP_CurrentFlowRateSensor(coordinator),
+                DROP_CurrentSystemPressureSensor(coordinator),
+                DROP_TemperatureSensorC(coordinator),
+                DROP_TemperatureSensorF(coordinator),
+            ]
+        )
+    elif device_type == DEV_RO_FILTER:
+        entities.extend(
+            [
+                DROP_FilterCart1Sensor(coordinator),
+                DROP_FilterCart2Sensor(coordinator),
+                DROP_FilterCart3Sensor(coordinator),
+                DROP_InletTdsSensor(coordinator),
+                DROP_OutletTdsSensor(coordinator),
+            ]
+        )
+
+    async_add_entities(entities)
+
+
+class DROP_CurrentFlowRateSensor(DROP_Entity, SensorEntity):
+    """Monitors the current water flow rate."""
+
+    _attr_icon = FLOW_ICON
+    _attr_native_unit_of_measurement = "gpm"
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _attr_translation_key = "current_flow_rate"
+
+    def __init__(self, device) -> None:
+        """Initialize the current flow rate sensor."""
+        super().__init__(self._attr_translation_key, device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the current flow rate."""
+        if self._device.current_flow_rate is None:
+            return None
+        return round(self._device.current_flow_rate, 1)
+
+
+class DROP_PeakFlowRateSensor(DROP_Entity, SensorEntity):
+    """Monitors the peak water flow rate for the day."""
+
+    _attr_icon = FLOW_ICON
+    _attr_native_unit_of_measurement = "gpm"
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _attr_translation_key = "peak_flow_rate"
+
+    def __init__(self, device) -> None:
+        """Initialize the peak flow rate sensor."""
+        super().__init__(self._attr_translation_key, device)
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the peak flow rate."""
+        if self._device.peak_flow_rate is None:
+            return None
+        return round(self._device.peak_flow_rate, 1)
+
+
+class DROP_WaterUsedTodaySensor(DROP_Entity, SensorEntity):
```

Comment:
Please don't mix Pascal_Case with CamelCase. I would prefer to see all CamelCase

Reply:
This was just a habit from an internal convention we use that is based on the fact that our brand name is all upper-case.  I removed the underscores in fed34201c0b635d9090600a083addabeed1df71f.